basket/news/tasks.py

```diff
@@ -201,6 +202,39 @@ def gmttime():
 
 
 @et_task
+def fxa_delete(data):
+    sfmc.upsert_row('FXA_Deleted', {'FXA_ID': data['uid']})
+
+
+@et_task
+def fxa_verified(data):
+    """Add new FxA users to an SFMC data extension"""
+    # used to be handled by the fxa_register view
+    email = data['email']
+    fxa_id = data['uid']
+    locale = data['locale']
+    subscribe = data.get('marketingOptIn')
+
+    # if we're not using the sandbox ignore testing domains
+    if email_is_testing(email):
+        return
+
+    lang = get_best_language(get_accept_languages(locale))
+    if not lang:
+        return
+
+    _update_fxa_info(email, lang, fxa_id)
+
+    if subscribe:
+        upsert_user.delay(SUBSCRIBE, {
+            'email': email,
+            'lang': lang,
+            'newsletters': settings.FXA_REGISTER_NEWSLETTER,
+            'source_url': settings.FXA_REGISTER_SOURCE_URL,
+        })
+
+
+@et_task
 def add_fxa_activity(data):
```

Comment:
@rfk after the other things, I believe the only other HTTP requests you send us are for this function which records account activity and includes the FxA ID, the `user_agent`, and a `first_device` boolean. Is this another event that will be on the queue?

Reply:
Yes, there will be a "login" event in the queue which we currently handle here:

  https://github.com/mozilla/fxa-basket-proxy/blob/master/lib/events/index.js#L86

Our handling of this event does two things.  It registers the activity by posting to `/fxa-activity/`:

  https://github.com/mozilla/fxa-basket-proxy/blob/master/lib/events/index.js#L114

And it also does a bit of magic where, if the user logged in with certain special values of the "utm_campaign" query parameter, it adds them to special subscriptions in basket:

  https://github.com/mozilla/fxa-basket-proxy/blob/master/lib/events/index.js#L94

It's a lightly hacky way for us to tunnel metrics info through FxA for various first-run-page experiments.

If you're happy to take over handling of those events as well, that'll be great!  We'll have to be careful though because IIUC the `/fxa-activity/` part is not idempotent.

> after the other things, I believe the only other HTTP requests you send us
> are for this function which records account activity

FWIW we also provide a "communication preferences" screen in FxA settings, where the user can change their choice of the marketing opt-in checkbox that they made during signup.  We currently action this by making calls to the existing basket web API, but we could consider whether there's a better way.

Reply:
> FWIW we also provide a "communication preferences" screen in FxA settings, where the user can change their choice of the marketing opt-in checkbox that they made during signup. We currently action this by making calls to the existing basket web API, but we could consider whether there's a better way.

That user interaction seems like a fine place for using the basket api, and if you're happy to keep doing it that's fine with me. I think we can move the activity handling to the queue, but that's a good point about it being different and not idempotent... Perhaps we should hold of on that bit and do it in a separate bug and sometime later?

Reply:
Filed #37 to cover this BTW.